pkg/kritis/review/review.go

```diff
@@ -103,15 +103,18 @@ func (r Reviewer) ReviewISP(images []string, isps []v1beta1.ImageSecurityPolicy,
 
 	for _, isp := range isps {
 		glog.Infof("Validating against ImageSecurityPolicy %s", isp.Name)
-		// Get all AttestationAuthorities in this policy.
-		auths, err := r.getAttestationAuthoritiesForISP(isp)
+		// Get the attestationauthority in this policy.
+		auth, err := r.getAttestationAuthorityForISP(isp)
 		if err != nil {
 			return err
 		}
+
 		for _, image := range images {
 			glog.Infof("Check if %s as valid Attestations.", image)
-			notAttestedBy := r.findUnsatisfiedAuths(image, auths)
-			imgAttested := len(notAttestedBy) == 0
+			imgAttested := false
+			if auth != nil {
```

Comment:
It should never be nil, right (based on L107-L110).  So perhaps this should return an error if auth == nil.

Reply:
There is a special case where AA name is empty string "". In this case, should we throw error much earlier on?

Reply:
You mean when isp.Name is empty?

Reply:
I mean isp.AttestationAuthorityName

Reply:
Oh, I think I understand now what is going on.  If you do not specify an AA in isp, then the isp decision is not "cached" as an attestation, and it is instead re-evaluated every time.  This sounds intentional, but I'm not sure it is documented anywhere.  We should just leave it as is, but document better.  WDYT?

Reply:
Sure. I'll update this PR with doc.